### PR TITLE
[New] Add loadCSS w/ git auto-update via single package.json

### LINKED ISSUE #7961: [Request] Add LoadCSS
**Library name:** LoadCSS
**Git repository url:** https://github.com/filamentgroup/loadCSS/
**npm package url(optional):** https://www.npmjs.com/package/fg-loadcss
**License(s):** MIT
**Official homepage:** https://github.com/filamentgroup/loadCSS

### PR DESCRIPTION
**Pull request by sashberd**
Pull request for issue: #7961
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/filamentgroup/loadCSS
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/fg-loadcss
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 219
   - Count of stars: 5416
   - Count of forks: 3161
 * NPM download stats (optional):
   - Downloads in the last day: 2154
   - Downloads in the last week: 12032
   - Downloads in the last month: 45634

# Essential checklist
 * [ ] I'm the author of this library
   * [X ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X ] This lib was not found on cdnjs repo
 * [X ] No already exist / duplicated issue and PR
 * [X ] The lib has notable popularity
   * [X ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X ] More than 800 downloads stats per month on npm registry
 * [X ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ X] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [ X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X ] The parent of the commit(s) in the PR is not old than 3 days.
 * [X ] Pull request is sending from a non-master branch with meaningful name.
 * [ X] Separate unrelated changes into different commits.
 * [X ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X ] Close corresponding issue in commit message
 * [ X] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@sufuf3 Would you like I will add folder v0.1.8 and files in it manually to next PR?

**Comment by sufuf3** [comment]
It sounds good, you can do this.
1. `npm i fg-loadcss@1.0.8` 
2. create a folder named `1.0.8` under `ajax/libs/loadcss/`
3. copy files which under `fg-loadcss/src` into `ajax/libs/loadcss/1.0.8/`

The commit message is 
`git commit -m "Manually add v1.0.8 of loadcss, cc ##7961" -m "Add v1.0.8 of the lib from npm becasue it is unavailable on GitHub"`

Thank you very much.

**Comment by sashberd** [comment]
No version1.0.8 on NPM, also v0.1.8 can be grabbed with latest pattern you aded

**Comment by sufuf3** [comment]
Yes you're right, I'll update the comment

**Comment by sufuf3** [comment]
@sashberd sorry let me check again

**Comment by sufuf3** [comment]
@sashberd I apologize.
We don't have to add v1.0.8 or v0.1.8.
There is some problem with my folder problem, I am fixing.

I'm sorry to bother you.
Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3 please help review this PR, thanks.

File: ajax/libs/loadcss/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "source": "git",
+    "target": "git://github.com/filamentgroup/loadCSS.git",
+    "fileMap": [
+      {
+        "basePath": "src",
+        "files": [
+          "*.js"
+        ]
+      },
+	  {
```
Comment (extend1994) [inline ajax/libs/loadcss/package.json]: Please remove unnecessary spaces here.

File: ajax/libs/loadcss/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "js"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/filamentgroup/loadCSS.git",
+    "fileMap": [
+      {
+        "basePath": "src",
+        "files": [
+          "*.js"
```
Comment (extend1994) [inline ajax/libs/loadcss/package.json]: Let's use `*load*` here. `*.js` will fetch all .js files which might involves those we don't need.

File: ajax/libs/loadcss/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        ]
+      },
+	  {
+        "basePath": "",
+        "files": [
+          "+(loadCSS|onloadCSS)*"
```
Comment (extend1994) [inline ajax/libs/loadcss/package.json]: How about `*load*` which is simpler?

**Comment by sashberd** [comment]
@extend1994 I installed web-minify-helper, but it is not working properly 
image

I had installed manually uglifyjs and clean-css-cli, but still same result. Also what option I should pass in order to create .map files
Could you provide detailed explanation, cause on official page it is not clear
Thanks

**Comment by sufuf3** [comment]
@sashberd 
As error message said, there is no files or dir.
Would you help me make sure that you executed `npm install` under `/webminifier/web-minify-helper`?
My packages are `clean-css-cli`  `uglify-es`  `uglify-js`, which under `/web-minify-helper/node_modules`.

Thank you.
cc @extend1994

**Comment by extend1994** [comment]
You could follow https://github.com/PeterDaveHello/web-minify-helper#how-to-use :smiley:

**Comment by sashberd** [comment]
@sufuf3 @extend1994 @PeterDaveHello 
I think web minifier is not working on Windows OS.
Here is my fs, the folders are in, but still same errors on minifier running:
image

Also there the URLS that you used as fallback of error should be using https
image
Another problem is mapping flag that is cannot be decalred. It is just not takes it to the args var.

I suggest to continue my PR via single package auto-update, as I did before
Waiting for you response

**Comment by PeterDaveHello** [comment]
@sashberd thanks for the updates and report! Do you have git bash installed on your Windows? Would you like to test if the minifier can run in that? Many thanks.

**Comment by sashberd** [comment]
@PeterDaveHello 
The printscreens above are my tests on windows with git bash. What another tests would you like to run?

**Comment by PeterDaveHello** [comment]
@sashberd Thanks, i may need to prepare a Windows environment myself to test it.

**Comment by sashberd** [comment]
@PeterDaveHello  I want to ask you team for finish this PR cause I do not have tools for this. Until you will fix minifier for windows. 
Also I will continue to create PR`s with single package json if you do not mind

**Comment by PeterDaveHello** [comment]
@sashberd we can take care of this one, but I wonder if you'd like to try something like Ubuntu on Windows 10 (not sure if you're using Win10) which may work? Prefer not to use single package.json too often, it actually make the review harder, thanks.

**Comment by sashberd** [comment]
@extend1994 @sufuf3 
The PR was updated

**Comment by sashberd** [comment]
@maruilian11 I was requested to change pattern by @extend1994. The previous pattern was nearly looks like you requested. So , to which pattern I suppose to update?

**Comment by maruilian11** [comment]
@sashberd Sorry, I think I mistook someting before.

**Comment by PeterDaveHello** [comment]
@sufuf3 I'm not very sure what you are doing and talking about, can you please discuss before doing any breaking changes? 

Point me if I'm wrong but comment is confusing me (or maybe it starts from the wrong commit which has difference content & commit message), what the things going on here isn't trivial at all, and I don't think you should abuse your permission to directly change so many things without any discussion, please stop doing this.

To finish this PR, please let me know what's going on here, what I saw here is @sashberd  add folder v1.0.0 with commit message saying v1.1.0, file `loadCSS.js` in v1.1.0 and v1.0.0 looks %100 the same(I didn't check the others yet), and @sufuf3 moved `1.0.0` to `1.1.0`.

**Comment by sashberd** [comment]
@sufuf3 Thanks for the fedback, You are right , it was my mistake with folder.
@PeterDaveHello Do not be angry in @sufuf3 👼 . 
In order to reduce mistakes i made new PR with right folder and files from NPM
FYR

**Comment by sufuf3** [comment]
Sorry about that.
I'll discuss before doing any breaking changes.

As your mention, it's right. I found 1.0.0 folder had be added, but actually the files under 1.0.0 is from npm. So I directly moved the 1.0.0 folder to 1.1.0 without any discuss before.

Really sorry about this.

---

@sashberd I think we can improve the second commit message from `Manually add loadcss@1.1.0` to `Manually add loadcss@1.1.0 from npm`, what do you think?
Because mention how did we get the files from the source can make us know more about the version of the lib.

And may I ask you how did you generate the minified files?
Did you just import the nom-minified files into the tool and generate it or just copy the content(the source code) of nom-minified files paste it on the tool and generate it?

Thank you.

**Comment by PeterDaveHello** [comment]
@sashberd I'm not angry, as she is under the training for more than one year, with powerful enough hardware/software resource support (and of course, it's all free, I support the whole training by my own), she has some responsibility on the review, the above issues are not what she supposed to do. She needs some reminders.

**Comment by sashberd** [comment]
@sufuf3 @PeterDaveHello The commit message was updated. FYR

**Comment by sufuf3** [comment]
@sashberd Thanks.
May I ask you how did you generate the minified files?
Did you just import the nom-minified files into the tool and generate it or just copy the content(the source code) of nom-minified files paste it on the tool and generate it?

Many thanks.

**Comment by sashberd** [comment]
@sufuf3 Inside the tool you have to provide folder and then select files and press compile. You could exam it your self  here http://koala-app.com/

**Comment by sufuf3** [comment]
ping @maruilian11 @extend1994 
Please help review this PR, thanks.

**Comment by sashberd** [comment]
@PeterDaveHello 
As for the web-minifier-helper problem, well I found the way to run your shell script from windows. The problem was that you script taking CWD from first provided parameter. In my case I run shell script as `./minify.sh path/to/the/files`, thus it will not found any node modules in current directory.
The solution was to change `MYPATH=$(dirname "$0")` to `MYPATH=$PWD`.
I have no idea if this change approprite for non windows platforms, but I think (I am not shell developer) you can check platform inside script and provide right directory to use.

**Comment by sufuf3** [comment]
ping @maruilian11 @extend1994

**Comment by maruilian11** [comment]
@PeterDaveHello Except the name, I think `fg-loadcss` or `loadCSS` would be better. I think everything is ok. BTW, I checked `1.0.0` and `1.1.0`, `loadCSS.js` and `onloadCSS.js` hashs are the same, but `cssrelpreload.js`'s is different. I think we can see them as different versions.

**Comment by sashberd** [comment]
@PeterDaveHello @maruilian11 The name of library was fixed. FYR

File: ajax/libs/loadCSS/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "source": "git",
+    "target": "git://github.com/filamentgroup/loadCSS.git",
+    "fileMap": [
+      {
+        "basePath": "src",
+        "files": [
+          "*load*"
+        ]
+      },
+	  {
```
Comment (sufuf3) [inline ajax/libs/loadCSS/package.json]: Let's use 6 spaces, replace 1 tab and 2 space.
```
-         {
+      {
```
Many thanks.

Comment (sashberd) [reply ajax/libs/loadCSS/package.json]: @sufuf3 I do not see such spaces.. However, I will run fixtool  agin and push

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

**Comment by sufuf3** [comment]
@sashberd Would you mind revising the second commit message?
From A to B
- A
```
Manually add loadCSS@1.1.0 from npm

Adding version from npm due to non existence on git repo, cc #7961, using web-minify-helper for minifing
```

- B
```
Manually add loadCSS@1.1.0 from npm

Adding version from npm due to non existence on git repo, cc #7961
using web-minify-helper for minifing
```

We follow the commit message 50/72 rule, you can refer to https://git-scm.com/book/en/v1/Distributed-Git-Contributing-to-a-Project. This document said
```
Short (50 chars or less) summary of changes

More detailed explanatory text, if necessary.  Wrap it to about 72
characters or so.  In some contexts, the first line is treated as the
subject of an email and the rest of the text as the body.  The blank
line separating the summary from the body is critical (unless you omit
the body entirely); tools like rebase can get confused if you run the
two together.
```
This is useful for maintainer and user use `git log` or check the [commits](https://github.com/cdnjs/cdnjs/commits/master).
What do you think?

Many thanks.

**Comment by sashberd** [comment]
@sufuf3 The PR was updated. FYR

**Comment by sashberd** [comment]
@extend1994 Please help me review this PR. Thanks

**Comment by sashberd** [comment]
@extend1994 I added some new keywords and removed map files from folder
As for the .gitignore, I never did such thing before and in order to avoid mistakes I want to ask you to help me with first time. Please add to .gitignore and I will see ho you did it.
Thanks for you help

**Comment by extend1994** [comment]
@sashberd You can check the updated PR :)
@sufuf3 Please help review modified parts, thanks.

**Comment by sashberd** [comment]
@extend1994 All is great. Thanks for you help

**Comment by sufuf3** [comment]
`Exclude v2.0.0-2 ~ v2.0.0-4 via .gitignore, cc #7961`
So, what the library be excluded by you?
@extend1994 Would you help to mention more clear, please?
Thank you.

**Comment by extend1994** [comment]
@sufuf3 Commit msg updated.

**Comment by sashberd** [comment]
@PeterDaveHello ping

**Comment by sashberd** [comment]
@PeterDaveHello ping

**Comment by PeterDaveHello** [comment]
I usually keep the duplicated versions, maybe we can just do nothing to them next time.
cc @extend1994 @sufuf3

File: ajax/libs/loadCSS/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "loadCSS",
+  "description": "A function for loading CSS asynchronously",
```
Comment (PeterDaveHello) [inline ajax/libs/loadCSS/package.json]: As you added v1.1.0 assets, it's better to add its version number here next time.

**Pull request merged**